### ISSUE TITLE
Viewport doesn't merge if another viewport is bellow main window

### ISSUE THREAD

**Issue by smilediver** [ocornut/imgui#7015]
If you have a main window and a platform window (B) bellow the main window, then if you move another platform window (A) above main window to merge it back - it doesn't work.

The problem is in `ImGui::UpdateTryMergeWindowIntoHostViewport()` since it iterates `g.Windows` from back to front and finds an overlap with viewport B, even though the B is bellow the main window and window A could be merged.

Video of above sitation:

![imgui_viewport_merge_bug](https://github.com/ocornut/imgui/assets/1411067/446b6c8f-02ea-4571-bde1-59a6ef1cf77d)

Version: v1.90-docking
Branch: docking
Running example_glfw_opengl3 on macOS.

**Comment by ocornut** [comment]
That's because we don't have a proper knowledge of what are the respective Z order of each platform window, we don't have access to this data.
Normally if you click "Dear ImGui Demo" once, then click your main app again, that info should be inferred again?

**Comment by smilediver** [comment]
No, changing window focuses doesn't help, because the actual issue is how `ImGui::UpdateTryMergeWindowIntoHostViewport()` checks for the overlaps.

If I understand correctly, the problem is that it's looking for any overlapping viewports in `g.Windows` and does the search back to front (Z orderish... see the nuances part). So in this case it will always find the overlap with the demo window and fail to merge. I think the correct way to check would be to start at the window being merged and do the front to back search. In such loop you'd need to check if window can be merged or it overlaps with the viewport being iterated.

Also a nuance to keep in mind is that `g.Windows` does sorting at the ImGui window level, but ignores viewports, ie it raises to front only the ImGui window being focused and not the other window being in the same viewport. So there can be windows from one viewport interspersed with another viewport's windows, and you need to be careful how you do the checks.

Btw, ImGui in this situation can correctly infer the viewport order as you can see in the metrics debugger (see inferred z order):

![image](https://github.com/ocornut/imgui/assets/1411067/7ba03ac5-8946-47f8-9215-a58cd8faa1eb)

Also the nuance thingy where debug log window appears to be below demo window, since `g.Windows` doesn't take into account the viewport.

![image](https://github.com/ocornut/imgui/assets/1411067/c23247da-303b-4d2b-bd32-b2db14f25009)

**Comment by ocornut** [comment]
Sorry for late answer. My understand is that this should now be fixed, the code for `UpdateTryMergeWindowIntoHostViewport()` has many issues back then. If somehow you are able to confirm that it'll be great, but I'll just assume it is fixed. The most recent changes include https://github.com/ocornut/imgui/issues/9128#issuecomment-3760626601
